scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -475,12 +481,29 @@ object Execution {
    */
 
   private trait ToWrite {
+    def setExtraConfig(conf: Config): Config = Config.empty
+
     def write(config: Config, flowDef: FlowDef, mode: Mode): Unit
   }
+
   private case class SimpleWrite[T](pipe: TypedPipe[T], sink: TypedSink[T]) extends ToWrite {
+
+    var extraConfig = Config.empty
```

Comment:
Yes the extraConfig var can grow in different moment see my unit test, we could have 

    val exec1 = TypedPipe.from(TypedTsv[(String, Int)](args("input1"))).writeExecution(TypedTsv[(String, Int)](args("output1")))
    val exec2 = TypedPipe.from(TypedTsv[(String, Int)](args("input2"))).writeExecution(TypedTsv[(String, Int)](args("output2")))
    exec1.setExtraConfig(Config(Map("testconf1" -> "1")))
    exec2.setExtraConfig(Config(Map("testconf2" -> "2")))
    val zippedExec = exec1.zip(exec2)
    zippedExec.setExtraConfig(Config(Map("testconf3" -> "3")))
    zippedExec.unit

So I expect to find testconf1 only in the flowStep generated by exec1, testconf2 only in the flowStep generated by exec2, and testconf3 in all the steps since it was set on the zipped execs.

I think to satisfy this use case we need to make use of that:

    var extraConfig = Config.empty